subsys/net/lib/lwm2m/lwm2m_message_handling.c

```diff
@@ -350,7 +352,7 @@ STATIC int build_msg_block_for_send(struct lwm2m_message *msg, uint16_t block_nu
 			LOG_ERR("coap packet init error: no output block context available");
 			return ret;
 		}
-		ret = coap_block_transfer_init(msg->out.block_ctx, lwm2m_default_block_size(),
+		ret = coap_block_transfer_init(msg->out.block_ctx, block_size,
```

Comment:
For this part, function `lwm2m_default_block_size()` should we slipt in 2 parts:
```suggestion
/* block start - replace lines 113-133*/
num coap_block_size lwm2m_default_block_size(void)
{
	return convert_to_coap_block_size_enum(CONFIG_LWM2M_COAP_BLOCK_SIZE);
}

static num coap_block_size convert_to_coap_block_size_enum(uint16_t block_size)
{
	switch (block_size) {
	case 16:
		return COAP_BLOCK_16;
	case 32:
		return COAP_BLOCK_32;
	case 64:
		return COAP_BLOCK_64;
	case 128:
		return COAP_BLOCK_128;
	case 256:
		return COAP_BLOCK_256;
	case 512:
		return COAP_BLOCK_512;
	case 1024:
		return COAP_BLOCK_1024;
	}

	return COAP_BLOCK_256;
}
/* block end */

		ret = coap_block_transfer_init(msg->out.block_ctx, convert_to_coap_block_size_enum(block_size),
```